novawallet/Modules/DApp/DAppBrowser/Tabs/DAppBrowserTabManager/DAppBrowserTabManager.swift

```diff
@@ -0,0 +1,369 @@
+
+            return tab
+        }
+        resultOperation.addDependency(saveTabOperation)
+
+        return CompoundOperationWrapper(
+            targetOperation: resultOperation,
+            dependencies: [saveTabOperation]
+        )
+    }
+
+    func retrieveWrapper(for tabId: UUID) -> CompoundOperationWrapper<DAppBrowserTab?> {
+        if let currentTab = tabs[tabId] {
+            return .createWithResult(currentTab)
+        } else {
+            let fetchTabOperation = repository.fetchOperation(
+                by: { tabId.uuidString },
+                options: RepositoryFetchOptions()
+            )
+
+            let resultOperation = ClosureOperation<DAppBrowserTab?> { [weak self] in
+                guard
+                    let self,
+                    let fetchResult = try fetchTabOperation.extractNoCancellableResultData()
+                else {
+                    return nil
+                }
+
+                return map(persistenceModel: fetchResult)
+            }
+
+            resultOperation.addDependency(fetchTabOperation)
+
+            return CompoundOperationWrapper(
+                targetOperation: resultOperation,
+                dependencies: [fetchTabOperation]
+            )
+        }
+    }
+
+    func updateRenderWrapper(
+        render: DAppBrowserTabRenderProtocol,
+        tab: DAppBrowserTab
+    ) -> CompoundOperationWrapper<Void> {
+        let renderDataOperation = render.serializationOperation()
+
+        let wrapper: CompoundOperationWrapper<Void> = OperationCombiningService.compoundNonOptionalWrapper(
+            operationManager: OperationManager(operationQueue: operationQueue)
+        ) { [weak self] in
+            guard let self else {
+                throw DAppBrowserTabManagerError.renderCacheFailed
+            }
+
+            let renderData = try renderDataOperation.extractNoCancellableResultData()
+
+            return saveRenderWrapper(
+                renderData: renderData,
+                tabId: tab.uuid
+            )
+        }
+
+        wrapper.addDependency(operations: [renderDataOperation])
+
+        return wrapper.insertingHead(operations: [renderDataOperation])
+    }
+
+    func removeTabWrapper(for tabId: UUID) -> CompoundOperationWrapper<Void> {
+        let deleteOperation = repository.saveOperation(
+            { [] },
+            { [tabId.uuidString] }
+        )
+
+        let renderRemoveWrapper = fileRepository.removeRender(for: tabId)
+        renderRemoveWrapper.addDependency(operations: [deleteOperation])
+
+        return renderRemoveWrapper.insertingHead(operations: [deleteOperation])
+    }
+
+    func removeAllWrapper() -> CompoundOperationWrapper<Void> {
+        let tabIds = tabs.map(\.value.uuid)
+
+        let rendersClearWrapper = fileRepository.removeRenders(for: tabIds)
+        let deleteOperation = repository.deleteAllOperation()
+
+        rendersClearWrapper.addDependency(operations: [deleteOperation])
```

Comment:
I don't see any reason to make dependency. Instead I would creating a result mapping operation (closure operation) that extracts results of both operations to make sure they completed successfully.